examples/permissioned-token/test/failing-paths.js

```diff
@@ -0,0 +1,627 @@
+const { types } = require('@algo-builder/runtime');
+const { assert } = require('chai');
+const { encodeAddress } = require('algosdk');
+const {
+  optInToASA,
+  optInToPermissionsSSC,
+  issue,
+  whitelist,
+  killToken,
+  transfer,
+  optOut,
+  forceTransfer,
+  setupEnv
+} = require('./common');
+
+const STR_TRANSFER = 'str:transfer';
+const RUNTIME_ERR1009 = 'RUNTIME_ERR1009: TEAL runtime encountered err opcode';
+const INDEX_OUT_OF_BOUND_ERR = 'RUNTIME_ERR1008: Index out of bound';
+const REJECTED_BY_LOGIC = 'RUNTIME_ERR1007: Teal code rejected by logic';
+
+describe('Permissioned Token Tests - Failing Paths', function () {
+  let runtime, master, alice, bob, elon;
+  let lsig, assetIndex, controllerAppID, permissionsAppId;
+  let asaDef, asaReserve, asaManager;
+
+  function syncAccounts () {
+    master = runtime.getAccount(master.address);
+    alice = runtime.getAccount(alice.address);
+    bob = runtime.getAccount(bob.address);
+    elon = runtime.getAccount(elon.address);
+  }
+
+  // Create Accounts and Env
+  function _setupEnv () {
+    [
+      runtime,
+      master,
+      alice,
+      bob,
+      elon,
+      assetIndex,
+      controllerAppID,
+      lsig,
+      permissionsAppId
+    ] = setupEnv();
+  }
+
+  this.beforeEach(async function () {
+    _setupEnv();
+    asaDef = runtime.getAssetDef(assetIndex);
+    asaReserve = runtime.getAccount(asaDef.reserve);
+    asaManager = runtime.getAccount(asaDef.manager);
+  });
+
+  describe('Token Issuance', function () {
+    it('should not issue token if receiver is not opted in', () => {
+      assert.throws(() =>
+        issue(runtime, asaReserve.account, elon, 20, controllerAppID, assetIndex, lsig),
+        `RUNTIME_ERR1404: Account ${elon.address} doesn't hold asset index ${assetIndex}`);
+    });
+
+    it('should not issue if token is killed', () => {
+      // Opt-in to ASA by receiver
+      optInToASA(runtime, elon.address, assetIndex);
+      syncAccounts();
+
+      killToken(runtime, asaManager.account, controllerAppID);
+      assert.throws(() =>
+        issue(runtime, asaReserve.account, elon, 20, controllerAppID, assetIndex, lsig),
+      RUNTIME_ERR1009);
+    });
+
+    it('should reject issuance tx if sender is not token reserve', () => {
+      // Opt-in to ASA by receiver
+      optInToASA(runtime, elon.address, assetIndex);
+      syncAccounts();
+
+      assert.throws(() =>
+        issue(runtime, bob.account, elon, 20, controllerAppID, assetIndex, lsig),
+      RUNTIME_ERR1009);
+    });
+
+    it('should reject issuance tx if trying to send asset using secret key instead of clawback', () => {
+      // Opt-in to ASA by receiver
+      optInToASA(runtime, elon.address, assetIndex);
+      syncAccounts();
```

Comment:
I think we don't need to call syncAccounts here (above as well)

Reply:
yes, checked. Removed `syncAccounts()` after optInToASA